app/grandchallenge/evaluation/models.py

```diff
@@ -1403,6 +1407,35 @@ def metrics_json_file(self):
             if output.interface.slug == "metrics-json-file":
                 return output.value
 
+    @property
+    def missing_metrics_json_file(self):
+        if self.status == self.SUCCESS:
```

Comment:
I think you would want to check `self.rank` here already. No need to fetch the outputs if the rank is not `0`

Reply:
Does the evaluation need to be public as well for this to be an issue? The issue description would suggest so. In that case, I would check that here as well.

Reply:
Indeed, I now check for self.rank before. The publish status would indeed affect the message, but I think the admin should still see a message about the missing metrics, and if published, an extended message that due to missing metrics the evaluation won't appear on the leaderboard.